karabo_bridge/server.py

```diff
@@ -0,0 +1,256 @@
+                 protocol_version='2.2', detector='AGIPD', raw=False,
+                 nsources=1, datagen='random', data_like='online', *,
+                 debug=True):
+        super().__init__(endpoint, sock=sock, protocol_version=protocol_version)
+
+        if ser != 'msgpack':
+            raise ValueError("Unknown serialisation format %s" % ser)
+
+        self.data = data_generator(
+            detector=detector, raw=raw, nsources=nsources, datagen=datagen,
+            data_like=data_like, debug=debug)
+        self.debug = debug
+
+    def loop(self):
+        print(f'Simulated Karabo-bridge server started on:\n'
+              f'tcp://{self.endpoint}')
+
+        timing_interval = 50
+        t_prev = time()
+        n = 0
+
+        for data, meta in self.data:
+            done = self.send(data, meta)
+            if done:
+                break
+
+            if self.debug:
+                print('Server : emitted train:',
+                      next(v for v in meta.values())['timestamp.tid'])
+            n += 1
+            if n % timing_interval == 0:
+                t_now = time()
+                print('Sent {} trains in {:.2f} seconds ({:.2f} Hz)'
+                      ''.format(timing_interval, t_now - t_prev,
+                                timing_interval / (t_now - t_prev)))
+                t_prev = t_now
+
+
+class ServerInThread(Sender):
+    def __init__(self, endpoint, sock='REP', maxlen=10, protocol_version='2.2',
+                 dummy_timestamps=False):
+        """ZeroMQ interface sending data over a TCP socket.
+
+        example::
+
+            # Server:
+            serve = ServerInThread(1234)
+            serve.start()
+
+            for tid, data in run.trains():
+                result = important_processing(data)
+                serve.feed(result)
+
+            # Client:
+            from karabo_bridge import Client
+            client = Client('tcp://server.hostname:1234')
+            data = client.next()
+
+        Parameters
+        ----------
+        endpoint: str
+            The address string.
+        sock: str
+            socket type - supported: REP, PUB, PUSH (default REP).
+        maxlen: int, optional
+            How many trains to cache before sending (default: 10)
+        protocol_version: ('1.0' | '2.1')
+            Which version of the bridge protocol to use. Defaults to the latest
+            version implemented.
+        dummy_timestamps: bool
+            Some tools (such as OnDA) expect the timestamp information to be in
+            the messages. We can't give accurate timestamps where these are not
+            in the file, so this option generates fake timestamps from the time
+            the data is fed in.
+        """
+        super().__init__(endpoint, sock=sock, protocol_version=protocol_version,
+                         dummy_timestamps=dummy_timestamps)
+        self.thread = Thread(target=self._run)
+        self.thread.daemon = True
```

Comment:
```suggestion
        self.thread = Thread(target=self._run, daemon=True)
```

This can be passed in the constructor since Python 3.3, according to the docs.

Reply:
Thanks, I did not know